numpy/__init__.cython-30.pxd

```diff
@@ -224,6 +298,12 @@ cdef extern from "numpy/arrayobject.h":
         # this field via the inline helper method PyDataType_SHAPE.
         cdef PyArray_ArrayDescr* subarray
 
+        @property
+        cdef inline npy_uint64 flags(self) nogil:
+            """The data types flags."""
+            return PyDataType_FLAGS(self)
```

Comment:
Does changing this to return an `npy_uint64` have any implications for e.g. running cython code compiled against numpy 2.x using a numpy 1.x wheel at runtime?

Reply:
OK, so that isn't a problem, but there is one problem:  If you use Cython 2 *and* access the highest flag, it is a negative number in Python (in 1.x), and that will fail flag checks.
Now that is only the aligned struct flag, and Cython 2, but will add a note.

We also need to cast to (usigned char) explicitly.